codepulse/src/main/scala/com/secdec/codepulse/util/ZipEntryChecker.scala

```diff
@@ -95,4 +86,33 @@ trait ZipEntryChecker {
 			stream.close
 		}
 	}
+
+	def findFirstEntry(file: File, recursive: Boolean = true)(callback: (String, ZipEntry, InputStream) => Boolean): Boolean = {
+		val stream = new BufferedInputStream(new FileInputStream(file))
+
+		try {
+			findFirstEntry(file.getName, stream, recursive)(callback)
+		} finally {
+			stream.close
+		}
+	}
+
+	def findFirstEntry(filename: String, stream: InputStream, recursive: Boolean)(callback: (String, ZipEntry, InputStream) => Boolean): Boolean = {
+		val zipStream = new ZipInputStream(stream)
+
+		try {
+			val entryStream = Stream.continually(Try { zipStream.getNextEntry })
+				.map(_.toOption.flatMap { Option(_) })
+		    	.takeWhile(_.isDefined)
+			    .flatten
+		    	.filterNot(ZipCleaner.shouldFilter)
+
+			entryStream.exists(entry =>
+				callback(filename, entry, zipStream)
+				||(recursive && isZip(entry.getName) && findFirstEntry(s"$filename/${entry.getName}", new CloseShieldInputStream(zipStream), true)(callback))
+			)
```

Comment:
Format nitpicks: 
 - Typically with a multi-line closure, you'd use curlies instead of parentheses, e.g. `{ entry => ... }`
 - Space after the `||`